AGENTS.md

```diff
@@ -0,0 +1,446 @@
+# Agent Guide for Yggdrasil Build Scripts
+
+This guide helps AI agents generate correct `build_tarballs.jl` recipes for BinaryBuilder.jl in the Yggdrasil repository.
+
+## Prerequisites
+
+- **BinaryBuilder.jl**: Requires Julia 1.7 specifically (use `julia +1.7` if juliaup is installed)
+- **Supported Platforms**: Linux (glibc and musl for x86_64, i686, aarch64, armv7l, armv6l, ppc64le, riscv64), Windows (x86_64, i686), macOS (x86_64, aarch64), FreeBSD (x86_64, aarch64)
+- Use `supported_platforms()` to get all available platforms
+
+## Special Dependencies
+
+Some dependencies require special handling:
+
+- **LLVM packages**: Must use `LLVM_full_jll` and match the version used by the Julia version. Requires careful ABI compatibility.
+- **MPI packages**: Need `MPIPreferences.jl` configuration and must use `MPItrampoline_jll` for cross-implementation compatibility.
+- **CUDA packages**: Require `CUDA_Runtime_jll` and must handle different CUDA versions. GPU code needs special compilation flags.
+
+For these complex dependencies, consult existing recipes in the repository (search for `LLVM_full_jll`, `MPItrampoline_jll`, or `CUDA_Runtime_jll`).
```

Comment:
Actually, for CUDA the packages probably don't have `CUDA_Runtime_jll` in their file, but instead they call `CUDA.required_dependencies` to get that.